src/plugin/message.cc

```diff
@@ -62,24 +63,37 @@ namespace chapel {
       printer->Print("\n");
     }
 
-    printer->Print("proc serialize(): bytes {\n");  
-    printer->Indent();
-    printer->Print("var s: bytes = b\"\";\n");  
-      
+    printer->Print("proc writeToOutputFile(ch) {\n");
+    printer->Indent(); 
+    
+    printer->Print(
+      "var binCh = new channel(writing=true, iokind.little, locking=false,\n"
```

Comment:
I was expecting that this code would be in the user-facing Chapel module supporting protobuf. Anyway it would be better if we can find a way to avoid putting it in the protobuf compiler plugin because that makes it harder to update and it's relatively strange code.

Put another way, in an expected use case, where is writeToOutputFile called? Is it possible to handle the locking and making the binary channel at that point?

If not, is it possible to put these details in a helper function?

Reply:
> Put another way, in an expected use case, where is writeToOutputFile called

`writeToOutputFile` is present in the chapel code generated through the plugin. It is called by the user in a `write.chpl` file (for e.g) to get the serialized values written to an `out` file. Example ->

```
use repeatedfield;
use IO;

var file = open("out", iomode.cw);
var writingChannel = file.writer();

var messageObj = new repeatedField();

messageObj.ui64.append(1);
messageObj.ui64.append(2445464);

messageObj.writeToOutputFile(writingChannel);
```

> I was expecting that this code would be in the user-facing Chapel module supporting protobuf.

I don't see a way of doing this with concrete functions, the `ch` channel from  `writeToOutputFile ` is passed directly to Append functions present in the user-facing Chapel module.

I also see that https://github.com/chapel-lang/chapel/pull/16032 is merged, can't we directly use the user exposed IO function for creating the new channel?

Reply:
> I also see that chapel-lang/chapel#16032 is merged, can't we directly use the user exposed IO function for creating the new channel?

Yes

> I don't see a way of doing this with concrete functions, the ch channel from writeToOutputFile is passed directly to Append functions present in the user-facing Chapel module.

Does that mean you do see a way of doing it with generic functions?

(e.g. message.writeToOutputFile calls Encoding.helpWriteToOutputFile(message, channel) which calls message.doWriteToOutputFile(channel) ?)

Reply:
@mppf I made the changes in 382a749 and updated the output file in the pr message. Please review.

Reply:
The example output looks good. It will be relatively easy to add similar methods for reading from / writing to bytes using this mechanism!

Reply:
Yes! The output is generated from the plugin itself. I think now we are good to go with this pr?